src/main/webapp/app/shared/util/firebase/firebase-utils.tsx

```diff
@@ -825,15 +849,15 @@ export const getTreatmentStats = (treatment?: Treatment) => {
     dxLevels: {} as { [dxLevel in DX_LEVELS]: number },
     pxLevels: {} as { [pxLevel in PX_LEVELS]: number },
   };
-  if (isTxLevelPresent(treatment.level)) {
-    stats.txLevels[treatment.level] = 1;
+  if (isTxLevelPresent(treatment?.level)) {
+    stats.txLevels[treatment?.level] = 1;
```

Comment:
Since `isTxLevelPresent` check if treatment is undefined or not, then could we do the following?
```suggestion
  if (isTxLevelPresent(treatment?.level)) {
    stats.txLevels[treatment.level] = 1;
```